### PR TITLE
PR #272 continued

### PR DESCRIPTION
**Pull request by matthiasr**
Picking up #272 to get it over the finish line:

* [x] fix the build issues
* [x] convert panics to log+exit (missed this in initial review)

cc @mhartenbower

### CODE REVIEW

**Comment by matthiasr** [comment]
note for the changelog: the logger change is only mildly breaking – the valid values for `--logger.format` have changed (`[logfmt, json]`) and the `fatal` log level is no longer valid.

File: main.go [hunk 1]
```diff
@@ -248,12 +270,16 @@ func main() {
...
+			os.Exit(1)
 		}
 		if *dumpFSMPath != "" {
 			err := dumpFSM(mapper, *dumpFSMPath, logger)
 			if err != nil {
-				panic(fmt.Sprintf("Error dumping FSM: %s", err))
+				level.Error(logger).Log("msg", "error dumping FSM", "error", err)
+				// Failure to dump the FSM is an error (the user asked for it and it
+				// didn't happen) but not fatal (the exporter is fully functional
+				// afterwards).
```
Comment (brian-brazil) [inline main.go]: As an aside, should this always exit anyway? This sounds more like a tool than what a daemon should do in passing.

**Comment by brian-brazil** [comment]
:+1:

File: main.go (hunk 1 above)
Comment (matthiasr) [reply main.go]: hmm it could, yeah, although then I'd prefer to start a mapper CLI which could also do config validation.

Comment (matthiasr) [reply main.go]: cf. #263

**Pull request merged**